app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -177,14 +180,46 @@ private void setupRecorder() {
         recorder.setAudioSource(MediaRecorder.AudioSource.MIC);
         recorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
         recorder.setOutputFile(fileName);
-        recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
+        if (isHeAacEncoderSupported()) {
+            recorder.setAudioEncoder(MediaRecorder.AudioEncoder.HE_AAC);

```

Comment:
Did you confirm what happens if you set this to a random integer that isn't in the supported set? The isSupported method below seems like an OK approach, but if we can handle this through an exception it might be easier future proofing.

Reply:
ahh that just throws an invalid arg exception for random integers on the API level without checking for the availability of codec on device. I could not find any docs signalling from where exactly such an exception might be getting thrown in the lifecycle so thought it would not be a good idea to rely on that.

Reply:
Got it, that makes sense to me given that the contract of the method signature isn't super clear. This approach looks great, thanks!